app/models/drupal_node.rb

```diff
@@ -16,10 +16,28 @@ def validate(record)
 class DrupalNode < ActiveRecord::Base
   include NodeShared # common methods for node-like models
 
-  attr_accessible :title, :uid, :status, :type, :vid, :cached_likes, :comment, :path
+  attr_accessible :title, :uid, :status, :type, :vid, :cached_likes, :comment, :path, :slug
   self.table_name = 'node'
   self.primary_key = 'nid'
 
+  extend FriendlyId
+  friendly_id :pretty_url, use: [:slugged, :history]
+
+  def should_generate_new_friendly_id?
+    slug.blank? || title_changed?
+  end
+
+  def pretty_url
```

Comment:
Perhaps this should be called "friendly_id_generator" or "friendly_id_adaptor" as the string it returns is not actually used as a URL.